cpp/common/src/codingstandards/cpp/Literals.qll

```diff
@@ -28,3 +28,22 @@ class Utf16StringLiteral extends StringLiteral {
 class Utf32StringLiteral extends StringLiteral {
   Utf32StringLiteral() { this.getValueText().regexpMatch("(?s)\\s*U\".*") }
 }
+
+/**
+ * A literal resulting from the use of a constexpr
+ * variable, or macro expansion.
+ */
+class CompileTimeComputedIntegralLiteral extends Literal {
+  CompileTimeComputedIntegralLiteral() {
+    this.getUnspecifiedType() instanceof IntegralType and
+    not this.getUnspecifiedType() instanceof BoolType and
```

Comment:
oh , this also deprecates/duplicates some of the individual exclusions that were used in the query - like 
```
  // Exclude character literals - I think this one can now be removed based on your regex? double check me on that!
  not l instanceof CharLiteral and 
  // Exclude `nullptr`
  not l.getType() instanceof NullPointerType and
  // Exclude boolean `true` and `false` - this can definitely now be removed
  not l.getType() instanceof BoolType and
```

Reply:
No, we are stating that it should not be a compile time computed value, which doesn't include char literals and booleans. So if also don't want those, we need to keep the above conditions.

Reply:
er Im not saying to omit them, I am saying [they are duplicated checks](https://github.com/github/codeql-coding-standards/pull/538/files#diff-6bbf6c525fda0e915ad2ae7af53c73ad2da5f2febc18cbf59349808dc138eaa0R42) (the snip I pasted was from the query file, but I did not link the line number, my bad)

Reply:
Not sure I'm following. The `CompileTimeComputedIntegralLiteral` class is intended to find cases where a `constexpr int foo = 1;` variable or macro `#define FOO 1` is used in a template instantiation. While there is some duplication in the query and the library predicate, it is still needed if the predicate is used outside the query at some point.

What are you suggesting to do with the duplicate checks?

Reply:
nvm, I was blanking on thinking that they were redundant in the query, but they are not! I was thinking bc `CompileTimeComputedIntegralLiteral` is never a bool, that the literal in the query was never going to be a bool but that is not accurate logic bc not all literals (bools included) are `CompileTimeComputedIntegralLiteral` to start! my bad